src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -456,6 +469,12 @@ private void EnsureEmbeddableAttributeExists(EmbeddableAttributes attribute, Dia
             {
                 SetNeedsGeneratedAttributes(attribute);
             }
+
+            if ((attribute & (EmbeddableAttributes.NullableAttribute | EmbeddableAttributes.NullableContextAttribute)) != 0 &&
+                modifyCompilation)
```

Comment:
so do we expect that in batch compilation, something passes `modifyCompilation: true` here, and in SemanticModel or whatnot we don't get here or we pass `modifyCompilation: false`?

Reply:
I guess I'm really wondering when/why `modifyCompilation` would be false.

Reply:
It looks like we pass `modifyCompilation: false` in cases where we want to report diagnostics but don't expect to emit any attributes - in particular binding lambdas and local functions.

---
In reply to: [300809615](https://github.com/dotnet/roslyn/pull/37019#discussion_r300809615) [](ancestors = 300809615)